### PR TITLE
Add stream-api and generics tutorials under JAVA

### LINKED ISSUE #1664: [Feature Request]: Add more topics under java tutorial
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Add below topics
1. stream api
2. generics'

### Use Case

NA

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
## Related Issue

fixes #1664 

## Description

I have added stream api and generics topics under java tutorial.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

### CODE REVIEW

File: docs/java/generics/Generic Classes.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+id: Generic Classes
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Generic Classes.md:2]: replace `id: Generic Classes` to `id: generic-classes`

File: docs/java/generics/Generic Classes.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+description: In this tutorial, we will explore how to create and use generic classes in JAVA.
+---
+
+# Generic Classes
+
+- Generic classes in Java allow you to define classes with placeholders for types.
+
+- These placeholders can then be replaced with actual types when creating instances of the class. This provides flexibility and type safety.
+
+- To define a generic class, you use angle brackets < > to declare type parameters. These parameters can be used throughout the class definition.
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Generic Classes.md:16]: replace `< >` to ``` `< >` ``` or may be use `&lt;` `&gt;`

File: docs/java/generics/Generic methods.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+id: Generic methods
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Generic methods.md:2]: replace `id: Generic methods` to `id: generic-methods`

File: docs/java/generics/Introduction to Generics.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+id: introduction
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Introduction to Generics.md:2]: replace `id: introduction` to `id: introduction-generics`

File: docs/java/generics/Wildcards.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
+---
+id: Wildcards
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Wildcards.md:2]: Replace `id: Wildcards` to `id: wildcards-generics`

File: docs/java/stream-apis/How can create streams.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+---
+id: creation of streams
```
Comment (Ajay-Dhangar) [inline docs/java/stream-apis/How can create streams.md:2]: Replace `id: creation of streams` to `id: creation-of-streams`

File: docs/java/stream-apis/Introduction to stream apis.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+id: introduction
```
Comment (Ajay-Dhangar) [inline docs/java/stream-apis/Introduction to stream apis.md:2]: Replace `id: introduction` to `id: introduction-stream-apis`

File: docs/java/stream-apis/Parallel streams.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+---
+id: Parallel streams
```
Comment (Ajay-Dhangar) [inline docs/java/stream-apis/Parallel streams.md:2]: Replace `id: Parallel streams` to `id: parallel-streams`

File: docs/java/stream-apis/Stream operations.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
+---
+id: Stream operations
```
Comment (Ajay-Dhangar) [inline docs/java/stream-apis/Stream operations.md:2]: Replace `id: Stream operations` to `id: stream-operations`

File: docs/java/stream-apis/Using stream with custom objects.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
+---
+id: Using stream with custom objects
```
Comment (Ajay-Dhangar) [inline docs/java/stream-apis/Using stream with custom objects.md:2]: Replace `id: Using stream with custom objects` to `id: using-stream-with-custom-objects`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now, update your code based on suggestions

**Comment by DharshiBalasubramaniyam** [comment]
> Now, update your code based on suggestions

Done! sorry for the inconvenience.

File: docs/java/generics/Wildcards.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+# Wildcards
+
+- wildcard types are used to make generic types more flexible, particularly when dealing with collections.
+
+- They allow you to define generic types that can accept a wider range of subtypes. Wildcard types are represented using the ? symbol.
+
+- There are two main types of wildcard bounds:
+
+## Upper Bounded Wildcards (<? extends T>):
```
Comment (Ajay-Dhangar) [inline docs/java/generics/Wildcards.md:18]: replace `<? extends T>` to ``` `<? extends T>` ```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
change code based on suggestions for only onr error

File: docs/java/generics/Wildcards.md (hunk 11 above)
Comment (DharshiBalasubramaniyam) [reply docs/java/generics/Wildcards.md]: done!

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**